homeassistant/components/cast/media_player.py

```diff
@@ -476,8 +486,20 @@ def media_seek(self, position):
 
     def play_media(self, media_type, media_id, **kwargs):
         """Play media from a URL."""
-        # We do not want this to be forwarded to a group
-        self._chromecast.media_controller.play_media(media_id, media_type)
+        if media_type == CAST_DOMAIN:
+            try:
+                app_data = json.loads(media_id)
+            except json.JSONDecodeError:
+                _LOGGER.error("Invalid JSON in media_content_id")
+                raise
+            app_name = app_data.pop("app_name")
+            try:
+                quick_play(self._chromecast, app_name, app_data)
+            except NotImplementedError:
+                _LOGGER.error("App %s not supported", app_name)
+        else:
+            # We do not want this to be forwarded to a group
```

Comment:
Move this comment before `if media_type == CAST_DOMAIN:`, it's a reminder that `play_media` will be interpreted as always targetting the specifi cast, unlike media controls which may be forwarded to a speaker group